src/query.rs

```diff
@@ -211,38 +211,77 @@ mod tests {
+                let indexed_crate_old = VersionedIndexedCrate::new(&crate_old);
+
+                let adapter =
+                    VersionedRustdocAdapter::new(&indexed_crate_new, Some(&indexed_crate_old))
+                        .expect("could not create adapter");
+                let results_iter = adapter
+                    .run_query(&semver_query.query, semver_query.arguments.clone())
+                    .unwrap();
+                (
+                    format!("./test_crates/{}/", crate_pair),
+                    results_iter
+                        .map(|res| res.into_iter().map(|(k, v)| (k.to_string(), v)).collect())
+                        .collect::<Vec<BTreeMap<_, _>>>(),
+                )
+            })
+            .filter(|(_crate_pair, output)| !output.is_empty())
             .collect();
 
         // Reorder both vectors of results into a deterministic order that will compensate for
         // nondeterminism in how the results are ordered.
-        let key_func = |elem: &BTreeMap<String, FieldValue>| {
-            (
-                elem["span_filename"].as_str().unwrap().to_owned(),
-                elem["span_begin_line"].as_usize().unwrap(),
-            )
+        let sort_individual_outputs = |results: &mut BTreeMap<String, Vec<_>>| {
+            let key_func = |elem: &BTreeMap<String, FieldValue>| {
+                (
+                    elem["span_filename"].as_str().unwrap().to_owned(),
+                    elem["span_begin_line"].as_usize().unwrap(),
+                )
+            };
+            for (_key, value) in results.iter_mut() {
+                value.sort_unstable_by_key(key_func);
+            }
         };
-        expected_results.sort_unstable_by_key(key_func);
-        actual_results.sort_unstable_by_key(key_func);
-
-        assert_eq!(expected_results, actual_results);
+        sort_individual_outputs(&mut expected_results);
+        sort_individual_outputs(&mut actual_results);
+
+        if expected_results != actual_results {
+            let results_to_string = |name, results| {
+                format!(
+                    "{}:\n{}",
+                    name,
+                    ron::ser::to_string_pretty(&results, ron::ser::PrettyConfig::default())
+                        .unwrap()
+                )
+            };
+            let messages = vec![
+                format!("Query {} produced incorrect output (./src/lints/{}.ron).", &query_name, &query_name),
+                results_to_string(format!("Expected output (./test_outputs/{}.output.ron)", &query_name), &expected_results),
+                results_to_string("Actual output".to_string(), &actual_results),
+                "Note that the individual outputs might have been deliberately reordered. Also, remember about running ./scripts/regenerate_test_rustdocs.sh when needed.".to_string(),
+            ];
+            panic!("\n{}\n", messages.join("\n\n"));
```

Comment:
My reasoning is that this whole message is huge and without the endlines it wraps with other lines from the `panic!` macro. With endlines it is more clear where the custom message starts and ends. Though I'm not sure whether it is a good solution, do you think I should remove the endlines? (to decide, it might be good to deliberately launch this `panic!` to see the output in the terminal)

Reply:
We can leave it as-is for now and focus on getting this merged. We can always change it later if we decide to.